internal/pkg/runtime/engine/apptainer/prepare_linux.go

```diff
@@ -166,6 +167,10 @@ func (e *EngineOperations) PrepareConfig(starterConfig *starter.Config) error {
 	// determine if engine need to propagate signals across processes
 	e.checkSignalPropagation()
 
+	// register internal image drivers
+	userNS, _ := namespaces.IsInsideUserNamespace(os.Getpid())
+	driver.RegisterImageDrivers(userNS, e.EngineConfig.File)
```

Comment:
You don't need to call RegisterImageDriver there, just conditionally set the default image driver via `e.EngineConfig.File.ImageDriver = "squashfsfuse"`, so the master process would load it